lib/puppet/pops/types/type_parser.rb

```diff
@@ -140,6 +140,7 @@ def self.type_map
         'iterable'     => TypeFactory.iterable,
         'iterator'     => TypeFactory.iterator,
         'string'       => TypeFactory.string,
+        'sensitive'    => TypeFactory.sensitive,
```

Comment:
I implemented it like so, making the parameterized type optional:

```ruby
    when 'sensitive'
      if parameters.size == 0
        TypeFactory.sensitive
      elsif parameters.size == 1
        param = parameters[0]
        assert_type(ast, param) unless param.is_a?(String)
        TypeFactory.sensitive(param)
      else
        raise_invalid_parameters_error('Sensitive', '0 to 1', parameters.size)
      end
```

What do you think of this approach?